test/unittest/create_test.sh

```diff
@@ -0,0 +1,62 @@
+#! /bin/bash
+
+# First argument needs to be the name of the script.
+
+usage_message="Usage: ./create_test.sh [--dir directory] my_test"
+if [[ $# -eq 0 || $# -gt 3 ]]; then
+    echo $usage_message
+    exit 128
+elif [[ $1 == "-d" && $# -ne 3 ]]; then
+    echo $usage_message
+    exit 128
+elif [[ $1 == "-d" && $# -eq 1 ]]; then
+    echo $usage_message
+    exit 128
+fi
+
+# If configured, grab the test directory name.
+test_dir=""
+if [[ $1 == "-d" ]]; then
+    echo $1
+    test_dir="$2"
+    shift ; shift ;
+fi
+
+# Check the test name
+if [[ $1 =~ ^[a-z][_a-z0-9]+$ ]]; then
+    echo "Generating test: $1..."
+else
+    echo "Invalid test name. Test name $1 must match the regex '[a-z][_a-z0-9]+$'"
+    exit 128
+fi
+
+# Check if the test already exists. Modify for all recrusive dirs
```

Comment:
You should also search for conflicting tests at the top level, we can assume that tests in different modules with the same name aren't conflicting. As long as the test suite specified in the test itself doesn't overlap.

Reply:
Done, let me know what you think of it.